glidertools/plot.py

```diff
@@ -350,16 +353,30 @@ def bin_size(depth, bins=None, ax=None, add_colorbar=True, **hist_kwargs):
         if bins is None:
             binning_freq = 50
             ybins = get_optimal_bins(depth, binning_freq)[0]
+            xbins = r_[nan, diff(ybins)]
+
```

Comment:
you can define `bins` as the `binning_freq` at the end of this if block. Then the second catch does not have to happen (see my later comment)